snap/snapcraft.yaml

```diff
@@ -0,0 +1,41 @@
+name: oidc-broker
+summary: OIDC Broker for authd
+description: |
+  Broker that enables OIDC authentication for authd.
+version: git
+grade: devel
+base: core24
+confinement: strict
+license: GPL-3.0+
+
+apps:
+  oidc-broker:
+    command: bin/oidc-broker
+    daemon: simple
+    slots:
+      - dbus-oidc
+    plugs:
+      - network
+      - config-files
+    restart-condition: always
+
+slots:
+  dbus-oidc:
+    interface: dbus
+    bus: system
+    name: com.ubuntu.authd.OidcBroker
+
+plugs:
+  config-files:
+    interface: system-files
+    read:
+      - /etc/authd/brokers.d/oidc-broker
+
+parts:
+  oidc-broker:
+    source: https://github.com/ubuntu/oidc-broker
+    source-branch: main
```

Comment:
can’t that be . to take the current branch? (and potentially, build snaps from other branches?)

Reply:
I'll answer both into the same comment, since they are related: yeah, we could leave the source-branch empty or use the local source to build the snap. I didn't do it for two reasons:
1) I think it's good to be as explicit as possible here to avoid confusion or misuse later;
2) When building the snaps through CI, we probably want to use the main branch instead of local files, no? (Especially since we might want to run the build in a separate branch)

Reply:
well, I see 1. being rather a source of confusion when hacking locally. And having worked as a dev advocate for snaps, I can ensure you this will happen :)

On 2. Let me check, but I think the snapstore is working by targetting a branch and checking it out, let me find back some old snaps config I have to confirm